src/main/java/Duke.java

```diff
@@ -1,10 +1,68 @@
+import commands.Command;
+import exceptions.DukeException;
+import parser.Parser;
+import storage.Storage;
+import tasklist.TaskList;
+import ui.Ui;
+
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    Duke() {
+        this.storage = new Storage();
+        this.ui = new Ui();
+        this.storage = new Storage();
+
+        ui.preload();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            if (e.getMessage().equals("LoadFileError")) {
+                ui.loadFileError();
+            }
+        }
+    }
+
+    /**
+     * Runs the program.
+     */
+    public void run() {
+        ui.introduction();
+        boolean isExit = false;
+        while (!isExit) {
```

Comment:
Hi, I like how clean this bit of code looks!

Reply:
Thanks!!